### PR TITLE
Add a CompositePageTransition to Avalonia.Visuals

### PR DESCRIPTION
**Pull request by worldbeater**
## What does the pull request do?

This PR adds a `<CompositePageTransition />` animation class that implements `IPageTransition` and allows us to execute multiple page transitions at once. This concept is quite similar to a `TransitionCollection` available in UWP. Using this class, one can declare a ReactiveUI `RoutedViewHost` control like this:

```xaml
<reactiveUi:RoutedViewHost Router="{Binding Router}">
   <reactiveUi:RoutedViewHost.PageTransition>
       <CompositePageTransition>
           <PageSlide Duration="0.5" />
           <CrossFade Duration="0.5" />
       </CompositePageTransition>
   </reactiveUi:RoutedViewHost.PageTransition>
</reactiveUi:RoutedViewHost>
```

And this results in the following good-looking transition that is a combination of `PageSlide` and `CrossFade`:

## What is the current behavior?

Currently, there is no way to use multiple transitions with a `RoutedViewHost` control. Also, during a conversation with @danwalmsley, we discovered that the `PageSlide` transition has a flicker effect, and combining `PageSlide` and `CrossFade` can be a possible way to avoid that.

## What is the updated/expected behavior with this PR?

Now, there is a `CompositePageTransition` that can be declared in XAML.

## Checklist

- [ ] Added unit tests (if possible)?
- [x] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Avaloniaui.net with user documentation

### CODE REVIEW

**Comment by worldbeater** [comment]
Added `SlideIn(Out)Easing` and `FadeIn(Out)Easing` properties to `CrossFade` and `PageSlide` transitions to make the setup even more flexible. These properties can be declared in XAML, so, for example, the following markup:
```xaml
<reactiveUi:RoutedViewHost Router="{Binding Router}">
    <reactiveUi:RoutedViewHost.PageTransition>
        <CompositePageTransition>
            <CrossFade Duration="0.4">
                <CrossFade.FadeInEasing>
                    <LinearEasing />
                </CrossFade.FadeInEasing>
                <CrossFade.FadeOutEasing>
                    <LinearEasing />
                </CrossFade.FadeOutEasing>
            </CrossFade>
            <PageSlide Duration="0.4" Orientation="Vertical">
                <PageSlide.SlideInEasing>
                    <CircularEaseInOut />
                </PageSlide.SlideInEasing>
                <PageSlide.SlideOutEasing>
                    <CircularEaseInOut />
                </PageSlide.SlideOutEasing>
            </PageSlide>
        </CompositePageTransition>
    </reactiveUi:RoutedViewHost.PageTransition>
</reactiveUi:RoutedViewHost>
```
Results in the following behavior:



And this markup:

```xaml
<reactiveUi:RoutedViewHost Router="{Binding Router}">
    <reactiveUi:RoutedViewHost.PageTransition>
        <CompositePageTransition>
            <CrossFade Duration="0.4">
                <CrossFade.FadeInEasing>
                    <LinearEasing />
                </CrossFade.FadeInEasing>
                <CrossFade.FadeOutEasing>
                    <LinearEasing />
                </CrossFade.FadeOutEasing>
            </CrossFade>
            <PageSlide Duration="0.5" Orientation="Vertical">
                <PageSlide.SlideInEasing>
                    <BounceEaseInOut />
                </PageSlide.SlideInEasing>
                <PageSlide.SlideOutEasing>
                    <CircularEaseInOut />
                </PageSlide.SlideOutEasing>
            </PageSlide>
        </CompositePageTransition>
    </reactiveUi:RoutedViewHost.PageTransition>
</reactiveUi:RoutedViewHost>
```

Produces the following:

File: src/Avalonia.Visuals/Animation/CompositePageTransition.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    /// </code>
+    /// </para>
+    /// </remarks>
+    public class CompositePageTransition : IPageTransition
+    {
+        /// <summary>
+        /// Gets or sets the transitions to be executed. Can be defined from XAML.
+        /// </summary>
+        [Content]
+        public List<IPageTransition> Transitions { get; set; } = new List<IPageTransition>();
```
Comment (jmacato) [inline src/Avalonia.Visuals/Animation/CompositePageTransition.cs:36]: `Transitions` by itself is a "reserved" keyword for Avalonia, let's rename this to `PageTransitions`

**Review by jmacato — CHANGES_REQUESTED** [review]
all looks good except for a nit :) great stuff! @worldbeater :)

**Review by jmacato — APPROVED** [review]
LGTM!

**Pull request merged**